src/namespaces/debug.rs

```diff
@@ -1,576 +1,39 @@
-use crate::{
-    fork::ForkSource,
-    node::{InMemoryNodeInner, MAX_TX_SIZE},
-    utils::{create_debug_output, storage_view::StorageView, to_real_block_number},
-};
-use itertools::Itertools;
 use jsonrpc_core::{BoxFuture, Result};
-use multivm::interface::VmInterface;
-use multivm::vm_latest::{constants::ETH_CALL_GAS_LIMIT, CallTracer, Vm};
-use once_cell::sync::OnceCell;
-use std::sync::{Arc, RwLock};
-use zksync_basic_types::H256;
-use zksync_core::api_server::web3::backend_jsonrpc::{
-    error::into_jsrpc_error, namespaces::debug::DebugNamespaceT,
-};
+use jsonrpc_derive::rpc;
 use zksync_types::{
-    api::{BlockId, BlockNumber, DebugCall, ResultDebugCall, TracerConfig, TransactionVariant},
-    l2::L2Tx,
+    api::{BlockId, BlockNumber, DebugCall, ResultDebugCall, TracerConfig},
     transaction_request::CallRequest,
-    PackedEthSignature, Transaction, U64,
+    H256,
 };
-use zksync_web3_decl::error::Web3Error;
-
-/// Implementation of DebugNamespaceImpl
-pub struct DebugNamespaceImpl<S> {
-    node: Arc<RwLock<InMemoryNodeInner<S>>>,
-}
 
-impl<S> DebugNamespaceImpl<S> {
-    /// Creates a new `Debug` instance with the given `node`.
-    pub fn new(node: Arc<RwLock<InMemoryNodeInner<S>>>) -> Self {
-        Self { node }
-    }
-}
-
-impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> DebugNamespaceT
-    for DebugNamespaceImpl<S>
-{
+#[rpc]
+pub trait DebugNamespaceT {
+    #[rpc(name = "debug_traceBlockByNumber")]
     fn trace_block_by_number(
         &self,
         block: BlockNumber,
         options: Option<TracerConfig>,
-    ) -> BoxFuture<Result<Vec<ResultDebugCall>>> {
-        let only_top = options.is_some_and(|o| o.tracer_config.only_top_call);
```

Comment:
Oh wow, we must have missed this file during the refactor when the `namespaces` directory was created. Thank you!